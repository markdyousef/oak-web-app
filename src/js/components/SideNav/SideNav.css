:root {
    --primary-color: #5ed892;
    --main-black-color: #131517;
    --light-black-color: #2d3438;
    --dark-gray-color: #737a80;
    --main-gray-color: #f2f2f2;
    --light-gray-color: #fafafa;
    --main-white-color: #fff;
}

.container {
    width: 200px;
    background-color: var(--light-black-color);
    padding: 25px;
    display: flex;
    flex-direction: column;
    align-items: baseline;
}

.collections {
    margin-top: 80px;
    height: 150px;
}

.collections h1 {
    color: var(--main-white-color);
    font-size: 16px;
    font-weight: bold;
}

.collections a {
    color: var(--light-gray-color);
    font-size: 16px;
    font-weight: lighter;
}

.subCategories div {
    height: 25px;
    display: flex;
    align-items: center;
}
