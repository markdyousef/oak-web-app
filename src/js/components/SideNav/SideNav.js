// @flow
import React, { Component, PropTypes } from 'react';
import { Link } from 'react-router';
import styled from 'styled-components';
import colors from '../../styles/colors';
import claiLogo from '../../../img/clai-logo.png';
// import HomeIcon from '../../icons/home';
import CollectionIcon from '../../icons/collections';

const Container = styled.nav`
    max-width: 230px;
    width: 100%;
    background-color: ${colors.darkGrey};
    display: flex;
    flex-direction: column;
    align-items: baseline;
    height: 100%;
    height: 100vh;
    position: relative;
    z-index: 10;
`;

const Team = styled.div`
    width: 100%;
    display: flex;
    align-items: center;
    padding: 16px;
`;

const TeamPic = styled.img`
    height: 40px;
    width: 40px;
    border-radius: 999em;
`;

const TeamName = styled.h1`
    color: ${colors.white};
    margin-left: 8px;
    font-weight: bold;
    font-size: 16px;
`;

const Nav = styled.section`
    margin-top: 24px;
    width: 100%;
`;
const StyledLink = styled(Link)`
    display: block;
    margin-bottom: 16px;
    color: rgba(255, 255, 255, 0.5);
    font-size: 15px;
    border-left: 3px solid ${colors.darkGrey};
    padding-left: 16px
`;

const LinkWrapper = styled.div`
    display: flex;
    align-items: center;
    & svg {
        margin-right: 8px;
        fill: ${colors.white};
    }
`;

const SubLink = styled(StyledLink)`
    margin: 14px;
    padding-left: 35px !important;
    /*width: 100%;*/
`;

export const Loading = styled.div`
    background-color: #393F43;
    height: 12px;
    width: ${props => props.width}%;
    border-radius: 2px;
`;

type Grove = {
    name: string,
    id: string
}
type DefaultProps = {};
type Props = {
    setUpdate: (update: bool) => void,
    shouldUpdate: bool,
    data: {
        loading: bool,
        groves?: Array<Grove>,
        refetch: Function
    }
};
type State = {};

class SideNav extends Component<DefaultProps, Props, State> {
    static defaultProps:DefaultProps;
    props: Props;
    state: State;
    constructor() {
        super();
        this.state = {};
    }
    componentWillReceiveProps(nextProps: Props) {
        const { data } = this.props;
        if (nextProps.shouldUpdate && !data.loading) {
            data.refetch();
            nextProps.setUpdate(false);
        }
    }
    renderSubCategories = () => {
        const { data: { loading, groves } } = this.props;
        const loadingCategories = 6;

        if (loading) {
            return (
                <div>
                    {[...Array(loadingCategories).keys()]
                        .map(item => (
                            <SubLink key={item + 'loading'}>
                                <Loading
                                    width={(Math.random() * (80 - 50)) + 50}
                                />
                            </SubLink>
                        ))
                    }
                </div>
            );
        }

        if (!loading && groves) {
            return (
                <div>
                    {groves.map(category => (
                        <div key={category.id}>
                            <SubLink
                                to={`collection/${category.id}`}
                                activeStyle={{ color: colors.white }}
                            >
                                {category.name}
                            </SubLink>
                        </div>
                    ))}
                </div>
            );
        }
        return null;
    }
    render() {
        return (
            <Container>
                <Team>
                    <TeamPic src={claiLogo} alt="team logo" />
                    <TeamName>Clai</TeamName>
                </Team>
                <Nav>
                    {/* <StyledLink
                        // to="/"
                        activeStyle={{ color: colors.white, borderColor: colors.green }}
                    >
                        <LinkWrapper><HomeIcon /><h1>Home</h1></LinkWrapper>
                    </StyledLink> */}
                    <StyledLink
                        to="/"
                        activeStyle={{ color: colors.white, borderColor: colors.green }}
                    >
                        <LinkWrapper><CollectionIcon /><h1>Collections</h1></LinkWrapper>
                    </StyledLink>
                    {this.renderSubCategories()}
                    {/* <StyledLink
                        // to="/"
                        activeStyle={{ color: colors.white, borderColor: colors.green }}
                    >
                        <div />
                        <h1>Scoreboards</h1>
                    </StyledLink>
                    <StyledLink
                        // to="/"
                        activeStyle={{ color: colors.white, borderColor: colors.green }}
                    >
                        <div />
                        <h1>Toolset</h1>
                    </StyledLink> */}
                </Nav>
            </Container>
        );
    }
}

export default SideNav;
