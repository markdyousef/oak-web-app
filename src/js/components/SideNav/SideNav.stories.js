import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import SideNav from './SideNav';

storiesOf('SideNav', module)
    .add('SideNav', () => (
        <SideNav
            data={{}}
        />
    ));
