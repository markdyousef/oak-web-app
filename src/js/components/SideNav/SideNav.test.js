import React from 'react';
import renderer from 'react-test-renderer';
import SideNav from './SideNav';

it('renders correctly', () => {
    const props = {
        data: {}
    };
    const component = renderer.create(
        <SideNav {...props} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
