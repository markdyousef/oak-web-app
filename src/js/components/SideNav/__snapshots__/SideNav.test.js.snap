exports[`test renders correctly 1`] = `
<nav
  className="eZvhkE">
  <div
    className="hHxprC">
    <img
      alt="team logo"
      className="cPvMma"
      src="test-file-stub" />
    <h1
      className="jUOHnK">
      Clai
    </h1>
  </div>
  <section
    className="cuAJPI">
    <a
      className="kKCtOZ"
      onClick={[Function]}
      style={Object {}}>
      <div
        className="Duegn">
        <svg
          height="16"
          viewBox="0 0 16 16"
          width="16">
          <g>
            <path
              d="M5.756,7.156H2.044c-0.772,0-1.4-0.628-1.4-1.4V2.043c0-0.772,0.628-1.4,1.4-1.4h3.712c0.772,0,1.4,0.628,1.4,1.4v3.713 C7.156,6.528,6.528,7.156,5.756,7.156z M2.044,1.644c-0.217,0-0.4,0.183-0.4,0.4v3.713c0,0.217,0.184,0.4,0.4,0.4h3.712 c0.221,0,0.4-0.179,0.4-0.4V2.043c0-0.221-0.179-0.4-0.4-0.4H2.044z" />
          </g>
          <g>
            <path
              d="M13.956,7.156h-3.712c-0.772,0-1.4-0.628-1.4-1.4V2.043c0-0.772,0.628-1.4,1.4-1.4h3.712c0.772,0,1.4,0.628,1.4,1.4v3.713 C15.356,6.528,14.729,7.156,13.956,7.156z M10.244,1.644c-0.217,0-0.4,0.183-0.4,0.4v3.713c0,0.217,0.184,0.4,0.4,0.4h3.712 c0.217,0,0.4-0.183,0.4-0.4V2.043c0-0.217-0.184-0.4-0.4-0.4H10.244z" />
          </g>
          <g>
            <path
              d="M5.756,15.356H2.044c-0.772,0-1.4-0.628-1.4-1.4v-3.713c0-0.772,0.628-1.4,1.4-1.4h3.712c0.772,0,1.4,0.628,1.4,1.4v3.713 C7.156,14.729,6.528,15.356,5.756,15.356z M2.044,9.844c-0.217,0-0.4,0.183-0.4,0.4v3.713c0,0.217,0.184,0.4,0.4,0.4h3.712 c0.221,0,0.4-0.179,0.4-0.4v-3.713c0-0.221-0.179-0.4-0.4-0.4H2.044z" />
          </g>
          <g>
            <path
              d="M13.956,15.356h-3.712c-0.772,0-1.4-0.628-1.4-1.4v-3.713c0-0.772,0.628-1.4,1.4-1.4h3.712c0.772,0,1.4,0.628,1.4,1.4 v3.713C15.356,14.729,14.729,15.356,13.956,15.356z M10.244,9.844c-0.217,0-0.4,0.183-0.4,0.4v3.713c0,0.217,0.184,0.4,0.4,0.4 h3.712c0.217,0,0.4-0.183,0.4-0.4v-3.713c0-0.217-0.184-0.4-0.4-0.4H10.244z" />
          </g>
        </svg>
        <h1>
          Collections
        </h1>
      </div>
    </a>
  </section>
</nav>
`;
